script.js

```diff
@@ -1,4 +1,120 @@
+};
+
+var choosePlayerScoreOrder = function (playerSelection) {
+  var playerScore = "";
+  if (playerSelection == "1") {
+    playerScore = String(currentPlayerRolls[0]) + String(currentPlayerRolls[1]);
+    allScores.push(playerScore);
+    myOutputValue = `Player ${currentPlayer}, you have selected ${playerSelection} and your score is ${playerScore}. `;
+    currentPlayerRolls = [];
+    gameMode = gameModeDiceRoll;
+    return myOutputValue;
+  } else if (playerSelection == "2") {
+    playerScore = String(currentPlayerRolls[1]) + String(currentPlayerRolls[0]);
+    allScores.push(playerScore);
+    myOutputValue = `Player ${currentPlayer}, you have selected ${playerSelection} and your score is ${playerScore}. `;
+    currentPlayerRolls = [];
+    gameMode = gameModeDiceRoll;
+    return myOutputValue;
+  } else {
+    myOutputValue = `Your input was invalid, please enter either "1" to use ${currentPlayerRolls[0]} or "2" to use ${currentPlayerRolls[1]} as the first number of the dice score.`;
+  }
+
+  return myOutputValue;
+};
+
+var resetGameModeClearScores = function () {
+  gameMode = gameModeDiceRoll;
+  currentPlayer = 1;
+  currentPlayerRolls = [];
+  allScores = [];
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+
+  if (currentPlayer == "1" && gameMode == "dice roll") {
+    myOutputValue = getPlayerHand();
+    gameMode = gameModeChooseOrder;
+    return myOutputValue;
+  }
+
+  if (currentPlayer == "1" && gameMode == "player choice") {
+    myOutputValue = choosePlayerScoreOrder(input);
+    currentPlayer = 2;
+    return (
+      myOutputValue +
+      "<br><br> It is now player 2's turn. Click submit to roll the dice, Player 2!"
+    );
+  }
+
+  if (currentPlayer == "2" && gameMode == "dice roll") {
+    myOutputValue = getPlayerHand();
+    gameMode = gameModeChooseOrder;
+    return myOutputValue + "";
+  }
+
+  if (currentPlayer == "2" && gameMode == "player choice") {
+    myOutputValue = choosePlayerScoreOrder(input);
+    gameMode = gameModeCompareScores;
+    return myOutputValue + "<br><br> click submit to see who's won!";
+  }
+
+  if (currentPlayer == "2" && gameMode == "compare scores") {
+    // resetGameModeClearScores();
+
+    myOutputValue =
+      "Player 1,  your score was " +
+      allScores[0] +
+      ".<br> Player 2, your score was " +
+      allScores[1] +
+      ".";
+
+    if (allScores[0] > allScores[1]) {
+      myOutputValue =
+        myOutputValue + "<br> Player 1 wins!😎<br> Press submit to play again.";
+    } else if (allScores[1] > allScores[0]) {
+      myOutputValue =
+        myOutputValue + "<br> Player 2 wins!😎<br> Press submit to play again.";
+    } else if ((allScores[0] = allScores[1])) {
+      myOutputValue =
+        myOutputValue + "<br> 🙌It's a tie!🙌<br> Press submit to play again.";
+    }
```

Comment:
Be very careful when comparing string values. In this case, the game works fine because both "numbers" are two digits. However consider the case of '2' vs '12'  which is the larger value? You may read more about this [here](https://www.w3schools.com/js/js_comparisons.asp#:~:text=When%20comparing%20a%20string%20with,NaN%20which%20is%20always%20false%20.&text=When%20comparing%20two%20strings%2C%20%222,1%20is%20less%20than%202.).